nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/kinesis/consumer/GetKinesis.java

```diff
@@ -0,0 +1,238 @@
+                context.getProperty(GetKinesis.KINESIS_STREAM_NAME).getValue(),
+                awsCredentialsProviderService.getCredentialsProvider(),
+                context.getProperty(GetKinesis.KINESIS_CONSUMER_WORKER_ID_PREFIX).getValue()+ ":"
+                    + InetAddress.getLocalHost().getCanonicalHostName() + ":" + UUID.randomUUID().toString())
+                    .withRegionName(context.getProperty(GetKinesis.REGION).getValue())
+                    .withInitialPositionInStream(
+                         InitialPositionInStream.valueOf(
+                             context.getProperty(KINESIS_CONSUMER_INITIAL_POSITION_IN_STREAM).getValue()))
+                    .withFailoverTimeMillis(
+                             context.getProperty(KINESIS_CONSUMER_DEFAULT_FAILOVER_TIME_MILLIS).asLong())
+                    .withMaxRecords(context.getProperty(KINESIS_CONSUMER_DEFAULT_MAX_RECORDS).asInteger())
+                    .withIdleTimeBetweenReadsInMillis(context.getProperty(KINESIS_CONSUMER_DEFAULT_IDLETIME_BETWEEN_READS_MILLIS).asLong())
+                    .withCallProcessRecordsEvenForEmptyRecordList(!context.getProperty(KINESIS_CONSUMER_DEFAULT_DONT_CALL_PROCESS_RECORDS_FOR_EMPTY_RECORD_LIST).asBoolean())
+                    .withParentShardPollIntervalMillis(context.getProperty(KINESIS_CONSUMER_DEFAULT_PARENT_SHARD_POLL_INTERVAL_MILLIS).asLong())
+                    .withShardSyncIntervalMillis(context.getProperty(KINESIS_CONSUMER_DEFAULT_SHARD_SYNC_INTERVAL_MILLIS).asLong())
+                    .withCleanupLeasesUponShardCompletion(context.getProperty(KINESIS_CONSUMER_DEFAULT_CLEANUP_LEASES_UPON_SHARDS_COMPLETION).asBoolean())
+                    .withTaskBackoffTimeMillis(context.getProperty(KINESIS_CONSUMER_DEFAULT_TASK_BACKOFF_TIME_MILLIS).asLong())
+                    .withMetricsBufferTimeMillis(context.getProperty(KINESIS_CONSUMER_DEFAULT_METRICS_BUFFER_TIME_MILLIS).asLong())
+                    .withMetricsMaxQueueSize(context.getProperty(KINESIS_CONSUMER_DEFAULT_METRICS_MAX_QUEUE_SIZE).asInteger())
+                    .withMetricsLevel(context.getProperty(KINESIS_CONSUMER_DEFAULT_METRICS_LEVEL).getValue());
+
+            batchSize = context.getProperty(GetKinesis.BATCH_SIZE).asInteger();
+            config.withMaxRecords(batchSize);
+
+            KinesisRecordProcessorFactory kinesisRecordProcessorFactory = new KinesisRecordProcessorFactory(this);
+            consumerWorker = new Worker
+                .Builder()
+                .recordProcessorFactory(kinesisRecordProcessorFactory)
+                .config(config)
+                .build();
+
+            executor.execute(consumerWorker);
+        } catch (Exception e) {
+            throw new ProcessException(e);
+        }
+    }
+
+    @OnShutdown
+    public void onShutdown() {
+        if (consumerWorker != null)
+            consumerWorker.shutdown();
+        executor.shutdownNow();
```

Comment:
@olegz  - The consumerWorker.shutdown() only sets a flag and does not throw an exception.  If I've missed anything, please advice.